app/controllers/concerns/authentication_and_sso_concerns.rb

```diff
@@ -89,6 +89,7 @@ def extend_session!
   def set_api_cookie!
     return unless @session_object
 
+    session.delete :value
```

Comment:
"It turns out, that when the app receives a cookie encrypted using the opposite key, it somehow adds a session item named :value with a value of an encrypted "null" string. Then, each time the cookies get read by the other app using the opposite secret_key_base, the “value” session keeps getting encrypted, which gets bigger in size, thus causing the error. While I haven’t found what causes this, fortunately, there’s a simple workaround. You just need to add a before_action filter to your base controller class that removes the problematic session." https://blog.kmkonline.co.id/rotating-ruby-on-rails-secret-key-base-9a4dcdf0d817